src/lib.rs

```diff
@@ -395,6 +420,93 @@ impl CspList {
+
+    Note that, while this algoritm is defined as operating on a global object, the only property it
+    actually uses is the global's CSP List. So this function operates on that.
+    */
+    pub fn does_sink_type_require_trusted_types(&self, sink_group: &str, include_report_only_policies: bool) -> bool {
+        let sink_group = &sink_group.to_owned();
+        // Step 1: For each policy in global’s CSP list:
+        for policy in &self.0 {
+            // Step 1.1: If policy’s directive set does not contain a directive whose name is "require-trusted-types-for", skip to the next policy.
+            let directive = policy.directive_set.iter().find(|directive| directive.name == "require-trusted-types-for");
+            // Step 1.2: Let directive be the policy’s directive set’s directive whose name is "require-trusted-types-for"
+            if let Some(directive) = directive {
+                // Step 1.3: If directive’s value does not contain a trusted-types-sink-group which is a match for sinkGroup, skip to the next policy.
+                if !directive.value.contains(sink_group) {
+                    continue;
+                }
+                // Step 1.4: Let enforced be true if policy’s disposition is "enforce", and false otherwise.
+                let enforced = policy.disposition == PolicyDisposition::Enforce;
+                // Step 1.5: If enforced is true, return true.
+                if enforced {
+                    return true;
+                }
+                // Step 1.6: If includeReportOnlyPolicies is true, return true.
+                if include_report_only_policies {
+                    return true;
+                }
+            }
+        }
+        // Step 2: Return false.
+        false
+    }
+    /**
+    https://w3c.github.io/trusted-types/dist/spec/#should-block-sink-type-mismatch
+
+    Note that, while this algoritm is defined as operating on a global object, the only property it
+    actually uses is the global's CSP List. So this function operates on that.
+    */
+    pub fn should_sink_type_mismatch_violation_be_blocked_by_csp(&self, sink: &str, sink_group: &str, source: &str) -> (CheckResult, Vec<Violation>) {
+        use CheckResult::*;
+        let sink_group = &sink_group.to_owned();
+        // Step 1: Let result be "Allowed".
+        let mut result = Allowed;
+        let mut violations = Vec::new();
+        // Step 2: Let sample be source.
+        let sample = source;
+        // Step 3: If sink is "Function", then:
+        if sink == "Function" {
+            // Step 3.1: If sample starts with "function anonymous", strip that from sample.
+            // Step 3.2: Otherwise if sample starts with "async function anonymous", strip that from sample.
+            // Step 3.3: Otherwise if sample starts with "function* anonymous", strip that from sample.
+            // Step 3.4: Otherwise if sample starts with "async function* anonymous", strip that from sample.
```

Comment:
What about these steps?

Reply:
I didn't implement them yet as there are no tests for these. There is no function sink group and I reckon that these steps might get deleted. Will file a ticket at the spec to ask what's up